### ISSUE TITLE
Preroll and main video play simultaneously on Android device, when main video is of type m3u8

### ISSUE THREAD

**Issue by tbourigault** [googleads/videojs-ima#1098]
When the video is started as soon as possible on an Android device and the main video is of type m3u8, the preroll and main video will start playing both at the same time, instead of playing the preroll first and then the main video. The preroll only is visible but we can hear the audio of the main video in the background.

I have made a copy of the simple example changing only the main video from a mp4 to a m3u8

original (mp4 no bug)
https://googleads.github.io/videojs-ima/examples/simple/ 

modified version (m3u8 bug)
https://tbourigault.github.io/IMA-SDK-Plugin-VideoJS-Simple-M3U8.html

**Steps to reproduce:**

- Open https://tbourigault.github.io/IMA-SDK-Plugin-VideoJS-Simple-M3U8.html on an **_Android_** device.
- Click on the play button of the video as soon as possible

**Expected behavior:**
The preroll ad should play first, followed by the main video.

**Actual behavior:**
Both the preroll ad and main video start playing at the same time. The preroll only is visible but we can hear the audio of the main video in the background.

Environment details:
Bug was reproduced on multiple android devices, different m3u8 files and versions of videoJS 7 & 8

one example:
Operating system: Android 6.0.1
Browser: Google Chrome (Version 106.0.5249)
Video player: video.js (Version 7.18.1) with the videojs-ima plugin (Version 2.0)

**Comment by Kiro705** [comment]
Hello @tbourigault 

Android 6.0.1 is marked as no longer maintained. If you are able to develop a fix using using your device, I can take a look at the pull-request. Otherwise, the plugin may not be able to support that version of Android.

Thank you,
Jackson
IMA SDK team

**Comment by tbourigault** [comment]
Hello @Kiro705 

I was able to reproduce it on more recent device also
2 person on my team were also able to reproduce the bug on:
Phone Samsung s22, android 13
Tablet Samsung Tab A7, Android 12

Thank you

**Comment by toperharley** [comment]
I have the same issue, really weird to hear the audio of the 2 videos at the same time.

@Kiro705 Any idea about this bug? must be a timing problem?

The important point is to click on the player as soon as the video icon appears, knowing that I can reproduce almost every time. Does not seem to happen when we wait a few seconds or more.

My phone is a Samsung Galaxy A51, Android 13.

**Comment by floverdevel** [comment]
Hi 👋🏻☺️
I was able to reproduce on me device.

Chrome 114.0.5735.60
Android 11; ONEPLUS A6013 Build/RKQ1.201217.002

Both audios can be heard simultaneously with this URL:
https://tbourigault.github.io/IMA-SDK-Plugin-VideoJS-Simple-M3U8.html

**Comment by Kiro705** [comment]
Hello all,

I was looking into this issue more, and have been able to reproduce it reliably, when making sure to play the video as soon as possible. Unfortunately, I have not been able to identify a fix for the issue, but will plan to take another look soon. Please let me know if any changes on your end seem to resolve this issue.

Thank you,
Jackson
IMA SDK team

**Comment by floverdevel** [comment]
> @floverdevel  how are you being able to play mid roll and post roll ads. as i am only being able to play pre-roll ads. can you please share the code.
> @Kiro705 

I never tried to play mid-roll or post-roll ads.

**Comment by davlasq** [comment]
Are there any plans of fixing it or workarounds? Maybe downgrade to some particular version?

**Comment by davlasq** [comment]
Here is my workaround, it doesn't solve the root issue but masks it for users:
`
let tempVolume;

player.ima.addEventListener(google.ima.AdEvent.Type.STARTED, () => {
  const adVideoEl = player.el().querySelector(".ima-ad-container video[src]");
  tempVolume = player.volume();
  player.volume(0);
  adVideoEl.volume = tempVolume;
});

player.ima.addEventListener(google.ima.AdEvent.Type.CONTENT_RESUME_REQUESTED, () => {
    player.currentTime(0);
    player.volume(tempVolume);
    player.src(player.src());
    player.play();
  }
);
`
So basically I mute video on ad start, then unmute and reload on ad end.

**Comment by sunilitgd** [comment]
Hi Everyone, Can you please advise this issue is fixed? Still I am facing this issue in VideoJS latest version

**Comment by Kiro705** [comment]
Hi all, sorry for the delay on this issue.

When looking into it today, I see the preroll ad start and finish playing, and then the content starts. The test URLs load the 2.4.0 version of the plugin. Could this issue have been resolved in a previous version?
